code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -361,65 +361,68 @@
 
 	return
 
+/mob/living/simple_animal/bot/medbot/proc/assess_beaker_injection(mob/living/carbon/C)
+	//If we have and are using a medicine beaker, return any reagent the patient is missing
+	if(use_beaker && reagent_glass?.reagents.total_volume)
+		for(var/datum/reagent/R in reagent_glass.reagents.reagent_list)
+			if(!C.reagents.has_reagent(R.id))
+				return R.id
+
+/mob/living/simple_animal/bot/medbot/proc/select_medication(mob/living/carbon/C, beaker_injection)
+	var/treatable_virus = FALSE
+	if(treat_virus)
+		for(var/thing in C.viruses)
+			var/datum/disease/D = thing
+			if(!(D.visibility_flags & HIDDEN_SCANNER && D.visibility_flags & HIDDEN_PANDEMIC) && D.severity != NONTHREAT && (D.stage > 1 || D.spread_flags & AIRBORNE))
```

Comment:
Style nitpick
```suggestion
			if(!(D.visibility_flags & (HIDDEN_SCANNER | HIDDEN_PANDEMIC)) && D.severity != NONTHREAT && (D.stage > 1 || D.spread_flags & AIRBORNE))
```

Reply:
Oh good point, this can actually be simplified even further

Reply:
..or maybe not because bitmath is hard :v
Your suggestion actually changes the check from "is the disease visible in any way (= not both flags at once)" to "is the disease completely visible (= no flags at all)"; I tried to come up with a solution that keeps the logic the same but I'm not sure it can be done with just bitmath besides doing an outright `==` comparison to both flags `|`'d together, which on the other hand would break down as soon as somebody adds a new visibility flag. Feel free to correct me if you can figure something out

Reply:
You’re right, I’ll try to come up with a proper solution soon. If I can’t, I’ll just call this resolved